### PR TITLE
[MMGIS] Modify Independent time contexts, open in new tab action, and expose overlay plots to support MMGIS pivoting

### LINKED ISSUE #5953: [MMGIS] Expose overlay plot to allow external plugins to import it
Exposing overlay plot so that it's available for import

### PR DESCRIPTION
**Pull request by khalidadil**
Closes #5953

### Describe your changes:
Changes described in source issue

### All Submissions:

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [X] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [X] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [X] Command line build passes?
* [X] Has this been smoke tested?
* [X] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/plugins/openInNewTabAction/openInNewTabAction.js [hunk 1]
```diff
@@ -31,8 +31,8 @@ export default class OpenInNewTab {
 
         this._openmct = openmct;
     }
-    invoke(objectPath) {
-        let url = objectPathToUrl(this._openmct, objectPath);
+    invoke(objectPath, urlParams = undefined) {
```
Comment (khalidadil) [inline src/plugins/openInNewTabAction/openInNewTabAction.js:34]: Allow new tab action to take custom url params

File: src/plugins/plot/MctPlot.vue [hunk 2]
```diff
@@ -383,10 +383,8 @@ export default {
         },
         setTimeContext() {
             this.stopFollowingTimeContext();
```
Comment (khalidadil) [inline src/plugins/plot/MctPlot.vue:385]: spacing cleanup

File: src/tools/url.js [hunk 3]
```diff
@@ -24,9 +24,27 @@
  * Module defining url handling.
  */
 
-export function paramsToArray(openmct) {
-    // parse urlParams from an object to an array.
+function getUrlParams(openmct, customUrlParams) {
```
Comment (khalidadil) [inline src/tools/url.js:27]: Merge router params with custom url params that are passed in

File: src/ui/components/components.js [hunk 4]
```diff
@@ -22,8 +22,10 @@
 
 import ObjectView from './ObjectView.vue';
 import StackedPlot from '../../plugins/plot/stackedPlot/StackedPlot.vue';
+import Plot from '../../plugins/plot/Plot.vue';
 
 export default {
     ObjectView,
-    StackedPlot
+    StackedPlot,
+    Plot
```
Comment (khalidadil) [inline src/ui/components/components.js:30]: Expose the overlay plot to plugins

File: src/api/time/IndependentTimeContext.js [hunk 5]
```diff
@@ -202,8 +202,13 @@ class IndependentTimeContext extends TimeContext {
     }
 
     getUpstreamContext() {
-        let timeContext = this.globalTimeContext;
+        const objectKey = this.openmct.objects.makeKeyString(this.objectPath[0].identifier);
+        const independentContext = this.globalTimeContext.independentContexts.get(objectKey);
+        if (independentContext) {
```
[the remark this answers is not in the document]
Comment (khalidadil) [reply src/api/time/IndependentTimeContext.js:207]: I worked with @shefalijoshi to find this solution for the time context code always returning the upstream context.

**Comment by davetsay** [comment]
LGTM but i think you are fixing a unit test

**Review by davetsay — APPROVED** [review]
LGTM

**Pull request merged**